code/modules/power/supermatter/supermatter.dm

```diff
@@ -1,396 +1,586 @@
+				if(powerloss_inhibitor < 0.5)
+					radio.autosay("DANGER: CHARGE INERTIA CHAIN REACTION IN PROGRESS.", name, "Engineering", list(z))
 
+			if(combined_gas > MOLE_PENALTY_THRESHOLD)
+				radio.autosay("Warning: Critical coolant mass reached.", name, "Engineering", list(z))
 
+		if(damage > explosion_point)
+			countdown()
+
+	return 1
+
+/obj/machinery/power/supermatter_crystal/bullet_act(obj/item/projectile/Proj)
+	var/turf/L = loc
+	if(!istype(L))
+		return FALSE
+	if(!istype(Proj.firer, /obj/machinery/power/emitter))
+		investigate_log("has been hit by [Proj] fired by [key_name(Proj.firer)]", "supermatter")
 	if(Proj.flag != "bullet")
 		power += Proj.damage * config_bullet_energy
 		if(!has_been_powered)
 			investigate_log("has been powered for the first time.", "supermatter")
-			message_admins("[src] has been powered for the first time <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>(JMP)</a>.")
-			has_been_powered = 1
-	else
+			message_admins("[src] has been powered for the first time [ADMIN_JMP(src)].")
+			has_been_powered = TRUE
+	else if(takes_damage)
 		damage += Proj.damage * config_bullet_energy
-	supermatter_zap()
-	return 0
+	return FALSE
 
-/obj/machinery/power/supermatter_shard/singularity_act()
+/obj/machinery/power/supermatter_crystal/singularity_act()
 	var/gain = 100
-	investigate_log("Supermatter shard consumed by singularity.","singulo")
-	message_admins("Singularity has consumed a supermatter shard and can now become stage six.<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>(JMP)</a>.")
+	investigate_log("Supermatter shard consumed by singularity.", "singulo")
+	message_admins("Singularity has consumed a supermatter shard and can now become stage six.")
 	visible_message("<span class='userdanger'>[src] is consumed by the singularity!</span>")
-	for(var/mob/M in GLOB.mob_list)
-		M << 'sound/effects/supermatter.ogg' //everyone gunna know bout this
-		to_chat(M, "<span class='boldannounce'>A horrible screeching fills your ears, and a wave of dread washes over you...</span>")
+	var/supermatter_sound = 'sound/effects/supermatter.ogg'
+	for(var/mob/M in GLOB.player_list)
+		if(M.z == z)
+			SEND_SOUND(M, supermatter_sound) //everyone goan know bout this
+			to_chat(M, "<span class='boldannounce'>A horrible screeching fills your ears, and a wave of dread washes over you...</span>")
 	qdel(src)
-	return(gain)
+	return gain
 
-/obj/machinery/power/supermatter_shard/attack_robot(mob/user as mob)
-	if(Adjacent(user))
-		return attack_hand(user)
+/obj/machinery/power/supermatter_crystal/blob_act(obj/structure/blob/B)
+	if(B && !isspaceturf(loc)) //does nothing in space
+		playsound(get_turf(src), 'sound/effects/supermatter.ogg', 50, 1)
+		damage += B.obj_integrity * 0.5 //take damage equal to 50% of remaining blob health before it tried to eat us
+		if(B.obj_integrity == B.max_integrity)
```

Comment:
Something tells me this is a wrong way to go about this. It used to be 
`B.health > 100`